src/Core/Echo.Concrete/SpanExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+
+namespace Echo.Concrete
+{
+    internal static class SpanExtensions
+    {
+        internal static bool All<T>(this Span<T> span, T value)
+        {
+            foreach (var item in span)
+            {
+                if (!item.Equals(value))
```

Comment:
This calls `object.Equals`. Does this not cause a lot of boxing if `T` is a value type (such as `byte`?)

Reply:
It will not, as you can see [here](https://sharplab.io/#v2:C4LglgNgPgAgTARgLAChUDsCGBbApgZwAdMBjXAAgFESALAewDoBhO9EgJ12F1QG9Vyg8mHTd2WCORgIAbFLjkAysXSUAHt3T4wrfAKH8UQ48NG5xmSdLkAjOnUkBBCBAA8AFQB8ACmA0w+EoqHp7kRJjoADTk7uQAbpYArrgAlPom5IYZGQBmdJykNOTeCezC3NimYSppRtnGWfXZYDnFAIRgFQyUAI6JlvglSam1TWNSAOzkOQO4ANzpTQC+qIv1MFPA7MkLdRkre+QHS0A===), a `constrained` instruction is being used to elide the boxing.

Reply:
TIL!